c10/CMakeLists.txt

```diff
@@ -52,6 +52,22 @@ target_compile_options(c10 PRIVATE "-DC10_BUILD_MAIN_LIB")
 if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
   target_compile_options(c10 PRIVATE "-fvisibility=hidden")
 endif()
+
+option(C10_USE_IWYU "Use include-what-you-use to clean up header inclusion" OFF)
+if(C10_USE_IWYU)
+  find_program(iwyu NAMES include-what-you-use)
+  if(iwyu)
+    set(iwyu_cmd
+        "include-what-you-use"
+        "-Xiwyu"
+        "--transitive_includes_only"
+        "-Xiwyu"
+        "--no_fwd_decls"
+        )
```

Comment:
Any reason not to use the mapping files in `tools/iwyu`?

Reply:
I didn't know their existence. Now I will use them

Reply:
Is iwyu documented in CONTRIBUTING.md ? if not, should we drop a line there at least pointing at which tools exist?

Reply:
After some though I think that we should avoid maintaining additional mapping files as possible. Our current aim is to remove unused inclusions for speeding up building and facilitating code refactor. We should not focus on IWYU clean in the present. Otherwise, the contributors would have to change several meta data for building in addition to the source code. These things are extra burdens to them and would be easily forgotten so that the mapping files would become outdated and useless. (tools/iwyu/c10.imp contains outdated instructions such as 
```
{ include: ["<c10/core/ScalarType.h>", public, "<c10/core/StorageImpl.h>", public] },
 ```)

The ultra solution to deal with header inclusion is of course C++20 modules.

Reply:
> Is iwyu documented in CONTRIBUTING.md ? if not, should we drop a line there at least pointing at which tools exist?

Not. IWYU support is still considered experimental. It may not be mature enough to be mentioned in  CONTRIBUTING.md